Bug fixes

Quesos-Nachos opened the pull request:
Bot will no longer attempt to deserialize warning messages if the cat API is down. Commented some stuff. Changed error levels on log messages. Disposed of CTS token.

Cisien commented on Modix/Modules/CatModule.cs:
```diff
@@ -40,6 +40,8 @@ public class CatModule : ModuleBase
             // Send the link
             await Context.Channel.SendMessageAsync(message);
+
+            cts.Dispose();
```
shouldn't this be called in a finally block?

Cisien replied:
or a using

Cisien commented on Modix/Services/Cat/CatService.cs:
```diff
@@ -16,6 +16,11 @@ public struct URL
+        /// Gets the cat GIF.
+        /// </summary>
+        /// <param name="token">The token.</param>
+        /// <returns>Return the URL to the cat</returns>
```
I smell stock StyleCop comments -- my opinion on these is to either update them to be more useful than the method name and parameters suggest, or remove them.

Cisien commented on Modix/Services/Cat/CatService.cs:
```diff
@@ -55,10 +74,19 @@ public struct URL
+                // json. If the website returns a blank string, something is wrong. Break the loop
                 if (string.IsNullOrWhiteSpace(json)) break;
 
                 //Deserialize the json retrieved from the website
```
I'm seeing doubles (this code is identical to the code above, and can probably be extracted to a common method

Cisien commented on Modix/Services/Cat/CatService.cs:
```diff
@@ -79,7 +107,13 @@ public struct URL
             return fileFoundFlag ? obj.file : "404 cat picture not found";
         }
```
I'm seeing doubles (this code is identical to the code above, and can probably be extracted to a common method

Quesos-Nachos commented:
I'm going to rewrite some of it to use Discord's built in dependency injection per Inzanit's suggestion. Going to add an interface and some other stuff.

Cisien commented on Modix/Modules/CatModule.cs:
```diff
@@ -3,39 +3,51 @@
-            else if (gif.Equals("gif", StringComparison.OrdinalIgnoreCase))
+                MediaType = (Media) Enum.Parse(typeof(Media), param, true);
+
+                if (!Enum.IsDefined(typeof(Media), MediaType) && !MediaType.ToString().Contains(","))
```
if enum is not defined, and it doesn't contain a comma? An enum would never contain a comma...?

Cisien commented on Modix/Services/Cat/CatService.cs:
```diff
@@ -5,81 +5,68 @@
+
+    public class CatService : ICatService
     {
         private static readonly HttpClient Client = new HttpClient();
```
I'd make this a private readonly member, and register the service as a singleton instead

Quesos-Nachos commented:
Registered service as Singleton, moved the enum, and removed the enum parsing logic. (I didn't like it. I think the **if** statements are easier to read)  The command can take a Media type parameter, but each command has to be either **!cat picture** or **!cat gif**. I wanted to retain functionality for **!cat** being the command to get pictures
